pkg/iam/iam.go

```diff
@@ -478,7 +481,8 @@ func (s *IAMService) DeleteRole() error {
 
 	// In a certain GiantSwarm release we changed how the IAM Roles are managed within `cluster-aws`. Crossplane will manage the roles from now on.
 	// This means that we no longer need to manage the IAM Roles for nodes (workers, control-plane) from this controller.
-	// If a cluster is using a release equal or greater than the release containing these changes, we skip the nodes IAM Role creation.
+	// If a cluster is using a release equal or greater than the release containing these changes, we skip the node IAM Role deletion.
+	// We have an issue to delete them manually when customers have upgraded https://github.com/giantswarm/giantswarm/issues/34712.
```

Comment:
We shouldn't have to manually delete stuff. That's what the operator code is for. We can enable deletion starting at a later release as long as we still have a finalizer on relevant objects – do we?

I'm putting this comment into the issue as well.